docs(applicationset): explain how to add repo credentials (#20453)
argoproj/argo-cd: docs/operator-manual/applicationset/Generators-Git.md

```

After saving, please restart the ApplicationSet pod for the changes to take effect.

> 

Reviewer: I did add a section regarding this in the [project](https://argo-cd.readthedocs.io/en/latest/user-guide/projects/) section (and for some reason it's not being displayed in the `stable` release 🤔), although it probably makes sense to have this in the Git generator section as well...

Author: Ah, look at that :)

Not sure where it should actually belong, i kinda find it weird to have under "project", since the applicationset can't belong to a project anyway.
For me it belongs under Generators-git, since that's where it will be used. But perhaps a mention directly on Applicationset is good too.

Author: I'm happy either way, I just wanted to write it down somewhere so people knew how it worked.
Up to you 👍

Reviewer: This makes sense to me 😄

## Repository credentials for ApplicationSets

> If your [ApplicationSets](index.md) uses a repository where you need credentials to be able to access it, you need to add the repoistory as an "non project scoped" repository.  

Reviewer: ```suggestion
If your [ApplicationSets](index.md) uses a repository where you need credentials to be able to access it, you need to add the repository as a "non project scoped" repository.  
```

> - In the UI, this is the "blank" value in the dropdown menu.

Reviewer: ```suggestion
- When doing that through the UI, set this to a **blank** value in the dropdown menu.
```

> - For CLI, make sure you DO NOT supply the parameter `--project` ([argocd repo add docs](../../user-guide/commands/argocd_repo_add.md))

Reviewer: ```suggestion
- When doing that through the CLI, make sure you **DO NOT** supply the parameter `--project` ([argocd repo add docs](../../user-guide/commands/argocd_repo_add.md))
```

> - For YAML spec, make sure you DO NOT have `project:` defined under `stringData:` ([complete yaml example](../argocd-repositories-yaml.md))

Reviewer: ```suggestion
- When doing that declaratively, make sure you **DO NOT** have `project:` defined under `stringData:` ([complete yaml example](../argocd-repositories-yaml.md))
```
